oeps/oep-0012.rst

```diff
@@ -0,0 +1,475 @@
+=================================
+OEP-12: Pluggable User Interfaces
+=================================
+
+.. This is the template to use when you start a new OEP.
+
++---------------+-------------------------------------------+
+| OEP           | :doc:`OEP-12 </oeps/oep-0012>`            |
++---------------+-------------------------------------------+
+| Title         | Pluggable User Interfaces                 |
++---------------+-------------------------------------------+
+| Last Modified | 2016-12-12                                |
++---------------+-------------------------------------------+
+| Author        | Andy Armstrong <andya@edx.org>            |
++---------------+-------------------------------------------+
+| Arbiter       | Clinton Blackburn <clinton@edx.org>       |
++---------------+-------------------------------------------+
+| Status        | Draft                                     |
++---------------+-------------------------------------------+
+| Type          | Architecture                              |
++---------------+-------------------------------------------+
+| Created       | 2016-12-06                                |
++---------------+-------------------------------------------+
+
+Abstract
+========
+
+There is a long standing desire to allow developers to extend edx-platform by
+providing plugins, rather than by modifying the platform's code directly.
+XBlocks provide this ability for new types of courseware components, but
+courseware is only one of many ways in which developers need to extend the
+platform (see `Feature Plugins for edX Platform`_ for a detailed enumeration).
+This proposal recommends using Django apps as the plugin mechanism, and
+providing a new `DjangoComponentView` to define components that can be
+consumed by Django or through a REST API.
+
+Motivation
+==========
+
+There are a number of driving use cases that act as motivation for supporting
+pluggable user interfaces.
+
+Use Case 1: Course Tabs
+-----------------------
+
+A very common need for new features is to be able to add a new course tab for
+the learner. Examples of this include Discussions, Student Notes, and Teams.
+There is a supported pluggable entry point for course tabs (see
+`Adding a new course tab`_) but this requires that the feature add a new URL
+for the feature, and render the whole page at that URL using a Mako template.
+It would be preferable for a course tab to be able to register a pluggable user
+interface that will be presented as a tab, and for the implementor not to need
```

Comment:
> implementor

This might be a British English thing - the more common spelling is implementer.